protocol-specs/eth2_spec_tests/tests/general/phase0/bls/verify/small/verify_infinity_pubkey_and_infinity_signature/data.yaml

```diff
@@ -0,0 +1,3 @@
+input: {pubkey: '0xc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000',
```

Comment:
this case must not be covered. And if our implementation is passing this test, then this means we skip an optimization. See [here](https://github.com/LiskHQ/lips-staging/blob/3a8960bbc8cfa5ca17b7ef4a83fb95d4ac0208a7/proposals/lip-0038.md#verify).

Reply:
We are not handling this case from our implementation. An error `BLST_PK_IS_INFINITY` is thrown from internal library and we catch and return `false` for any error. If its a compliance behaviour matching with eth2 as well then I suggest we should not try to patch the lib for it.

Reply:
> We are not handling this case from our implementation.
Well, LIP 0038 is excluding this case/scenario. But the implementation is actually (unnecessarily) prepared for it. The fact that an error is thrown is showing that `KeyValidate` is called. This behavior is not breaking anything, but it is highly recommended to not do it for performance reasons.

Maybe it's ok to leave it for now. But the impact of calling `KeyValidate` in every call to `Verify` on the performance should be investigated at some point.

Reply:
As discussed, we can leave for now. But it would be good to check the performance impact of doing the subgroup check in every call to `Verify`.